redisinsight/api/src/modules/shared/services/instances-business/overview.service.ts

```diff
@@ -192,27 +194,44 @@ export class OverviewService {
    * @param nodes
    * @private
    */
-  private calculateTotalKeys(nodes = []): number {
+  private calculateTotalKeys(nodes = []): [number, Record<string, number>] {
     if (!this.isMetricsAvailable(nodes, 'keyspace', [undefined])) {
-      return undefined;
+      return [undefined, undefined];
     }
 
     try {
       const masterNodes = filter(nodes, (node) => ['master', undefined].includes(
         get(node, 'replication.role'),
       ));
 
-      return sumBy(masterNodes, (node) => sum(
+      const totalKeysPerDb = {};
+
+      masterNodes.forEach((node) => {
         map(
           get(node, 'keyspace', {}),
-          (dbKeys): number => {
+          (dbKeys, dbNumber): void => {
             const { keys } = convertBulkStringsToObject(dbKeys, ',', '=');
-            return parseInt(keys, 10);
+
+            if (!totalKeysPerDb[dbNumber]) {
+              totalKeysPerDb[dbNumber] = 0;
+            }
+
+            totalKeysPerDb[dbNumber] += parseInt(keys, 10);
           },
-        ),
-      ));
+        );
+      });
+
+      // filter 0 values
+      Object.keys(totalKeysPerDb).forEach((key) => {
+        if (!totalKeysPerDb[key]) {
+          delete totalKeysPerDb[key];
+        }
+      });
+
+      const totalKeys = totalKeysPerDb ? sum(Object.values(totalKeysPerDb)) : undefined;
+      return [totalKeys, Object.keys(totalKeysPerDb).length > 1 ? totalKeysPerDb : undefined];
```

Comment:
May be will be clearer if move Object.keys(totalKeysPerDb).length > 1 ? totalKeysPerDb : undefined] to variable ?